From cccf7d0cb5dd6e6e66fce23577efffe1542fd43c Mon Sep 17 00:00:00 2001
From: Michael Himing <mhiming@gmail.com>
Date: Sun, 30 Apr 2017 19:01:07 +1000
Subject: [PATCH] Player lookup improvements v2

Paper's patch didn't change the get-by-uuid method, no idea why but here
you go!!

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 825e1346b..a6da7852e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -573,16 +573,10 @@ public final class CraftServer implements Server {
         // PaperSpigot end
     }
 
-    // TODO: In 1.8+ this should use the server's UUID->EntityPlayer map
     @Override
     public Player getPlayer(UUID id) {
-        for (Player player : getOnlinePlayers()) {
-            if (player.getUniqueId().equals(id)) {
-                return player;
-            }
-        }
-
-        return null;
+        EntityPlayer player = playerList.uuidMap.get(id);
+        return player != null ? player.getBukkitEntity() : null;
     }
 
     @Override
-- 
2.13.3

